### ISSUE TITLE
Portable Windows Version

### ISSUE THREAD

**Issue by RoyiAvital** [brrd/Abricotine#74]
Hello,

I'm not sure about the state of the current release.
But could we have a Portable Version for Windows?

What I mean by portable?

1. No Installation  
The software is delivered as archive and can decompressed into any folder and work fro there.
2. Local Settings  
The software keep all its data (settings, configuration, etc...) under the folder the program runs from.
3. System Privileges  
No need for any system privileges. The software only write and read from its folder. No service used by the program is running unless the program is running.

If we can get all those, it would be amazing.

Thank You.

P. S.
Is there a way to have MathJax functioning without Online Connection?
Namely, will it work on computers with no Internet access?
That could be a killer feature.

**Comment by brrd** [comment]
Normally 1 and 3 are already fulfilled, so it means the only thing to add to Abricotine is local settings, is it? 

>  Is there a way to have MathJax functioning without Online Connection?

This is already working offline, expect in HTML export where the script is downloaded from MathJax CDN (otherwise the HTML export would weigh more than 60 Mb).

**Comment by RoyiAvital** [comment]
@brrd , 
I guess you're correct.
Only to add version which writes locally instead of the User Folder.

Thank You.

P. S.
I wasn't aware it works offline, wow!
Could you feature to export to PDF offline?

**Comment by brrd** [comment]
Ok, so your request here is to add an option to store config directory under the program directory.
Config directory currently contains the config.json file, dictionaries and export templates.

**Comment by brrd** [comment]
>  Could you feature to export to PDF offline?

Yes, Pandoc support is planned (see #14).

**Comment by RoyiAvital** [comment]
@brrd ,

Yes, if all configuration files are under the same program file things will get portable.
It would be perfect.

Thank You.

**Comment by RoyiAvital** [comment]
Any update on this?

**Comment by RoyiAvital** [comment]
@brrd , Please, any option to have this?

Thank You.

**Comment by brrd** [comment]
Please @RoyiAvital stop asking again and again. If this issue is still open then you can guess that nobody worked on it yet. This feature is not in my priorities but I'm not the only developer here, anybody can contribute to this project.

**Comment by RoyiAvital** [comment]
I tried the approach given here - https://github.com/BoostIO/Boostnote/issues/344#issuecomment-318354371.  
Namely, I built a batch file (`Abricotine.bat`) to launch Abricotine and trick it on System Variables (`USERPROFILE`).

```
@echo off
if not exist "%CD%\profile\AppData\Roaming" mkdir "%CD%\profile\appdata\Roaming"
set USERPROFILE=%CD%\profile
if not exist "%CD%\profile\notes" goto message
goto start

:message
echo PORTABLE ABRICOTINE
echo -------------------------------
echo Please enter this storage path in Boostnote:  .\profile
echo.
pause 

:start
start Abricotine.exe /I
```

It didn't work:

![image](https://user-images.githubusercontent.com/7347975/36587826-e926a064-188e-11e8-8e2c-0394aa5bcb07.png)

@brrd ,
You referred me here.
I thought it means there are news.

Thank You.

**Comment by RoyiAvital** [comment]
OK,
I made a progress.
This batch file will make Abricotine launch:

```
@echo off
if not exist "%CD%\Profile\AppData\Roaming" mkdir "%CD%\Profile\AppData\Roaming"
if not exist "%CD%\Profile\Documents" mkdir "%CD%\Profile\Documents"
if not exist "%CD%\Profile\AppData\Local\Temp" mkdir "%CD%\Profile\AppData\Local\Temp"
set USERPROFILE=%CD%\Profile
set APPDATA=%CD%\Profile\AppData
set TEMP=%CD%\Profile\AppData\Local\Temp
if not exist "%CD%\profile\notes" goto message
goto start

:message
echo PORTABLE ABRICOTINE
echo -------------------------------
echo Please enter this storage path in Boostnote:  .\profile
echo.
pause 

:start
start Abricotine.exe /I
REM start Abricotine.exe
```

I have another problem which I'm not sure related to this.
When I try to launch preferences this is what I get:

![image](https://user-images.githubusercontent.com/7347975/36589199-382330de-1893-11e8-92f7-e182fba170d7.png)
